hazenlib/tasks/acr_geometric_accuracy.py

```diff
@@ -42,6 +45,65 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.ACR_obj = ACRObject(self.dcm_list)
 
+    def detect_best_rotation(self, ref_img):
+        """
+        Detects the best rotation/flip to align the ACR phantom automatically.
+        Returns a tuple: (rotation_name, rotation_function).
+        """
+        transforms = [
+            (lambda x: x, 'identity'),
+            (lambda x: np.rot90(x, 1), 'rot90'),
+            (lambda x: np.rot90(x, 2), 'rot180'),
+            (lambda x: np.rot90(x, 3), 'rot270'),
+            (lambda x: np.fliplr(x), 'fliplr'),
+            (lambda x: np.flipud(x), 'flipud'),
+            (lambda x: np.fliplr(np.rot90(x, 1)), 'rot90+fliplr'),
+            (lambda x: np.flipud(np.rot90(x, 1)), 'rot90+flipud'),
+        ]
+
+        ref_img = ref_img.astype(np.float64)
+        ref_img = (ref_img - ref_img.min()) / (ref_img.max() - ref_img.min() + 1e-12)
+
+        # Use first slice as reference.
+        mov_img = self.ACR_obj.slice_stack[0].pixel_array.astype(np.float64)
+        mov_img = (mov_img - mov_img.min()) / (mov_img.max() - mov_img.min() + 1e-12)
+
+        best_mse = np.inf
+        best_name = 'identity'
+        best_func = lambda x: x
+
+        for func, name in transforms:
+            transformed = func(mov_img)
+            # Align by centroid
+            mask_ref = ref_img > 0.1
+            mask_mov = transformed > 0.1
+            cy_ref, cx_ref = center_of_mass(mask_ref)
+            cy_mov, cx_mov = center_of_mass(mask_mov)
+            shift_y = cy_ref - cy_mov
+            shift_x = cx_ref - cx_mov
+            aligned = nd_shift(transformed, shift=(shift_y, shift_x), order=1, mode='constant', cval=0.0)
+
+            # Compute mean squared error.
+            min_shape = tuple(np.minimum(ref_img.shape, aligned.shape))
+            mse = np.mean((ref_img[:min_shape[0], :min_shape[1]] - aligned[:min_shape[0], :min_shape[1]]) ** 2)
+
+            if mse < best_mse:
+                best_mse = mse
+                best_name = name
+                best_func = func
+
+        logger.info(f"Best rotation/flip detected: {best_name} (MSE={best_mse:.4e})")
+        return best_name, best_func
+
+    def apply_rotation_to_stack(self, rotation_func):
```

Comment:
Similarly, it would be best if this was defined in one place rather than for every class. It also looks like there is some overlap with the method `rotate_images` in the `ACRObject.py` file.